executor/distsql.go

```diff
@@ -668,7 +668,7 @@ func (e *XSelectIndexExec) doIndexRequest() (distsql.SelectResult, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return distsql.Select(e.ctx.GetClient(), e.ctx.GoCtx(), selIdxReq, keyRanges, e.scanConcurrency, !e.outOfOrder, getIsolationLevel(sv), e.priority)
+	return distsql.Select(e.ctx.GetSessionVars().CurrentDB, e.ctx.GetClient(), e.ctx.GoCtx(), selIdxReq, keyRanges, e.scanConcurrency, !e.outOfOrder, getIsolationLevel(sv), e.priority)
```

Comment:
CurrentDB is the default database, but user could choose another database in their SQL. For example, `use test; ...; select * from test1.t;`.